lib/storage/index.native.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * On native we can leave almost everything to AsyncStorage, but files need to be prepared.
+ * Before files are saved in our key/value storage they should first be copied to
+ * our "local space" (App Documents) so that they're not lost - image pickers and document
+ * pickers would move files to cache which can be cleared before we've uploaded them
+ *
+ * We don't actually save the file to our key/value store but only the reference to it
+ */
+
+import Storage from './providers/AsyncStorage';
+import FileHandler from './providers/NativeFileHandler';
+
+const nativeStorage = {
+    ...Storage,
+
+    /**
+     * On native it's a no-op because it's not possible to have multiple instances of Onyx (e.g. multiple tabs)
+     */
+    synchronizeInstances() {},
+
+    /**
+     * On native we copy the file locally and return a reference to the new file
+     * This allows us to stringify and save the reference in our regular storage
+     * e.g. as part of a persisted request or an optimistic report action
+     * @param {{uri: string, name: string, type: string}} file
+     * @returns {{uri: string, name: string, type: string}}
+     */
+    prepareFile(file) {
+        // Todo: perhaps check size here and abort if the size is larger than 20MB
+
+        // Create result optimistically so the method can called as sync method
+        const uri = FileHandler.getLocalDestination(file);
+        const optimisticResult = {...file, uri};
+
+        // Todo: do something about the error, e.g. run eviction logic if it's due to lack of space
+        FileHandler.storeLocally(file)
+            .catch(console.error);
+
+        return optimisticResult;
+    }
```

Comment:
There's an alternative approach, where instead of having to call `prepareFile` we make a check every time we call `setItem` and recursively search the `value` for files:

https://github.com/Expensify/react-native-onyx/blob/86d8ad9f607254d33ef53df2bdf9d22786ff2e99/lib/storage/index.native.js#L25-L76